packages/widgets/src/health-monitoring/component.tsx

```diff
@@ -0,0 +1,319 @@
+"use client";
+
+import {
+  Avatar,
+  Box,
+  Card,
+  Center,
+  Flex,
+  Group,
+  HoverCard,
+  Indicator,
+  List,
+  Progress,
+  RingProgress,
+  Text,
+  Tooltip,
+} from "@mantine/core";
+import { useElementSize, useListState } from "@mantine/hooks";
+import {
+  IconBrain,
+  IconClock,
+  IconCpu,
+  IconCpu2,
+  IconFileReport,
+  IconInfoSquare,
+  IconServer,
+  IconTemperature,
+  IconVersions,
+} from "@tabler/icons-react";
+
+import type { HealthMonitoring } from "@homarr/integrations";
+import { useI18n } from "@homarr/translation/client";
+
+import type { WidgetComponentProps } from "../definition";
+import { NoIntegrationSelectedError } from "../errors";
+
+export default function HealthMonitoringWidget({
+  options,
+  integrationIds,
+  serverData,
+}: WidgetComponentProps<"healthMonitoring">) {
+  if (integrationIds.length === 0) {
+    throw new NoIntegrationSelectedError();
+  }
+  const t = useI18n();
+  const [healthData] = useListState<{
+    integrationId: string;
+    healthInfo: HealthMonitoring;
+  }>(serverData?.initialData ?? []);
+
+  return (
+    <Box h="100%">
+      {healthData.map(({ integrationId, healthInfo }) => {
+        const memoryUsage = formatMemoryUsage(healthInfo.memAvailable, healthInfo.memUsed);
+        const disksData = matchFileSystemAndSmart(healthInfo.fileSystem, healthInfo.smart);
+        const { ref, width } = useElementSize();
+        const ringSize = width * 0.95;
+        const ringThickness = width / 10;
+        const progressSize = width * 0.2;
+
+        return (
+          <Box h="100%">
+            <Card className="health-card" m="2.5cqmin" p="2.5cqmin" withBorder>
+              <Flex
+                className="health-card-elements"
+                h="100%"
+                w="100%"
+                justify="space-between"
+                align="center"
+                key={integrationId}
+              >
+                <Box className="health-card-info">
+                  <HoverCard>
+                    <HoverCard.Target>
+                      <Indicator
+                        inline
+                        processing
+                        color={healthInfo.rebootRequired ? "red" : healthInfo.availablePkgUpdates > 0 ? "blue" : "gray"}
+                        position="top-end"
+                        size="4cqmin"
+                        label={healthInfo.availablePkgUpdates > 0 ? healthInfo.availablePkgUpdates : undefined}
+                        disabled={!healthInfo.rebootRequired && healthInfo.availablePkgUpdates === 0}
+                      >
+                        <Avatar size="10cqmin" radius="sm">
+                          <IconInfoSquare size="10cqmin" />
```

Comment:
This icon looks too big to me. Do you agree?
Maybe make it slightly smaller?

Reply:
I resize it to "8cqmin". looks better now